src/python/gudhi/simplex_tree.pyx

```diff
@@ -797,6 +798,40 @@ cdef class SimplexTree:
         :rtype: bool
         """
         return dereference(self.get_ptr()) == dereference(other.get_ptr())
+    
+    def __getstate__(self):
+        """Pickle the SimplexTree data structure as a Python Byte Array
+        :raises MemoryError: In the case the serialization allocates a too large block of memory.
+        :returns: Serialized SimplexTree data structure
+        :rtype: numpy.array of shape (n,)
+        """
+        cdef size_t buffer_size = self.get_ptr().get_serialization_size()
+        # Let's use numpy to allocate a buffer. Will be deleted automatically
+        np_buffer = np.empty(buffer_size, dtype='B')
+        cdef char[:] buffer = np_buffer
+        cdef char* buffer_start = &buffer[0]
+        with nogil:
+            self.get_ptr().serialize(buffer_start, buffer_size)
+        
+        return np_buffer
+
+    def __setstate__(self, state):
+        """Construct the SimplexTree data structure from a Python Byte Array
+        :param state: Serialized SimplexTree data structure
+        :type state: numpy.array of shape (n,)
+        """
+        cdef char[:] buffer = state
+        cdef size_t buffer_size = state.shape[0]
+        # Backup old pointer
```

Comment:
I don't remember if I already asked: why only remove it at the end instead of immediately? I think a comment explaining it in the code would help.

Reply:
I moved the deletion and explained it on [3ad35bd](https://github.com/GUDHI/gudhi-devel/pull/810/commits/3ad35bd2a8dd3c4774133097c01420876cfebf5d)